packages/core/src/utils.ts

```diff
@@ -446,6 +446,43 @@ function deepEqualsObject(
   return true;
 }
 
+/**
+ * Checks if object2 includes all fields and values of object1.
+ * It doesn't matter if object2 has extra fields.
+ * @param object1 The first object.
+ * @param object2 The second object.
+ * @returns True if object2 includes all fields and values of object1.
+ */
+export function deepIncludes(object1: unknown, object2: unknown): boolean {
```

Comment:
Small nit on the naming: I usually try to match any verb in the name of a binary function to the relation between its arguments in order.  For example, this function signature says to me "object1 deepIncludes object2", when in reality it's testing the reverse.  One way around this would be to more clearly label the arguments: `object1` could be called `pattern` and `object2` just `value` or something like that.  What do you think of something like `deepMatches(value: any, pattern: any)`?